src/osd.c

```diff
@@ -364,6 +378,25 @@ render_osd(struct server *server, cairo_t *cairo, int w, int h,
 			case LAB_FIELD_TYPE:
 				buf_add(&buf, get_type(*view));
 				break;
+			case LAB_FIELD_WINFO:
+				if (rc.window_switcher.all_workspaces) {
+					buf_add(&buf, (*view)->workspace->name);
+					if ((*view)->maximized) {
+						buf_add(&buf, " M ");
+					} else if ((*view)->minimized) {
+						buf_add(&buf, " m ");
+					} else if ((*view)->fullscreen) {
+						buf_add(&buf, " F ");
+					} else {
+						buf_add(&buf, "   ");
+					}
+				}
```

Comment:
Should we really guard everything with `rc.window_switcher.all_workspaces` here?
The states (max, min, fullscreen) also make sense without the all_workspaces flag.

Similar thing for the output name below, that could have another `windowSwitcher` option like `showOutput`, thoughts @johanmalm ?

Reply:
I was originally leaving out the if all workspaces, and I have mixed feeling about producing the info
when only current workspace. But either way is really fine. 

The reason for short id vs long name, is just space saving. I didn't want to have so much info that
it would interfere with the next field (no trimming on type/winfo) and along with W/X, M/m/F is 
pretty much understood by most people having used a computer for any time.

Reply:
Alright, removed the if *all_workspaces. If they select "winfo" instead of "type" they will see
changes.